utils/sl/prereq_tools/base.py

```diff
@@ -366,28 +367,14 @@ def get(self, subdir, **kw):
 *********************** ERROR ************************\n""" % comp)
             raise DownloadFailure(self.url, subdir)
 
-        commands = ['git clone %s %s' % (self.url, subdir)]
-        if not RUNNER.run_commands(commands):
+        if (passed_commit_sha and not branch):
+            commands = ['git clone %s --branch %s --single-branch %s' % (self.url,
+                                                                         passed_commit_sha,
+                                                                         subdir)]
+            if not RUNNER.run_commands(commands):
+                raise DownloadFailure(self.url, subdir)
+        else:
             raise DownloadFailure(self.url, subdir)
```

Comment:
This looks to have built properly in CI, which means that "branch" is never set under any of the conditions we test, do we expect it to be?

Reply:
I believe we used to set branch for one of our components in the definition but it's probably not necessary since we can pass the sha/tag in utils/build.config.  We should probably just remove it.

Reply:
Agreed, but that's more than I'm willing to take on today.